stream_test.go

```diff
@@ -0,0 +1,179 @@
+package replicate_test
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/replicate/replicate-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStreamText(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprint(w, `event: output
+data: foo
+
+event: done
+
+`)
+	}))
+	defer ts.Close()
+
+	p := &replicate.Prediction{
+		URLs: map[string]string{
+			"stream": ts.URL,
+		},
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	c, err := replicate.NewClient(replicate.WithToken("test-token"))
+	require.NoError(t, err)
+
+	r, err := c.StreamPredictionText(ctx, p)
+
+	assert.NoError(t, err)
+
+	text, err := io.ReadAll(r)
+	assert.NoError(t, err)
+	assert.Equal(t, "foo", string(text))
```

Comment:
I don't follow why these are using `assert` and not `require` 🤔

Reply:
Yean we should use `require.NoError`, but we definitely want `assert.Equal`. I'm not very consistent but `require` is for when there's no point continuing the test.

```suggestion
	require.NoError(t, err)

	text, err := io.ReadAll(r)
	require.NoError(t, err)
	assert.Equal(t, "foo", string(text))
```